src/Query/QueryBuilder.php

```diff
@@ -309,14 +309,40 @@ public function fetchFirstColumn(): array
      */
     public function executeQuery(): Result
     {
+        [$cteParams, $cteParamsTypes] = $this->withQueriesParameters();
+
         return $this->connection->executeQuery(
             $this->getSQL(),
-            $this->params,
-            $this->types,
+            array_merge($cteParams, $this->params),
+            array_merge($cteParamsTypes, $this->types),
```

Comment:
These merges should happen only of there's at least one CTE that has parameters bound to it. With that said, I'd rework `withQueriesParameters()` to return the resulting parameters and types, not only the ones bound to CTEs.

Reply:
@morozov it's done